packages/x-components/src/x-modules/search-box/components/search-input-placeholder.vue

```diff
@@ -0,0 +1,294 @@
+<template>
+  <component
+    :is="animation"
+    v-if="isVisible"
+    class="x-input-placeholder x-search-input-placeholder"
+    mode="out-in"
+  >
+    <span :key="message" data-test="search-input-placeholder">
+      {{ message }}
+    </span>
+  </component>
+</template>
+
+<script lang="ts">
+  import Vue from 'vue';
+  import { Component, Prop, Watch } from 'vue-property-decorator';
+
+  import { animateTranslate, State, xComponentMixin, XOn } from '../../../components';
+  import { searchBoxXModule } from '../x-module';
+
+  /**.
+   * This component renders an animated placeholder for the search input in the shape of a list of
+   * iterating messages that can be configured to happen always or only when hovering the input
+   *
+   * @public
+   */
+  @Component({
+    mixins: [xComponentMixin(searchBoxXModule)]
+  })
+  export default class SearchInputPlaceholder extends Vue {
+    /**
+     * List of messages to animate.
+     */
+    @Prop({ required: true })
+    protected messages!: Array<string>;
+
+    /**
+     * Animation component used for the messages.
+     *
+     * @public
+     */
+    @Prop({ default: () => animateTranslate('bottom-to-top') })
+    protected animation!: Vue;
+
+    /**
+     * Time in milliseconds during which each message is visible.
+     *
+     * @public
+     */
+    @Prop({ default: 1500 })
+    protected animationIntervalMs!: number;
+
+    /**
+     * Whether the messages animation is active only when hovering the search input or always.
+     *
+     * @public
+     */
+    @Prop({ default: false })
+    protected animateOnlyOnHover!: boolean;
+
+    @State('searchBox', 'query')
+    public query!: string;
+
+    protected isSearchBoxHovered = false;
```

Comment:
I know that is verbose, but we should comment everything. Props, methods, variables, getters, computed...etc